Mixpanel/MPNotification.m

```diff
@@ -6,158 +6,97 @@
         
-        if (imageURLComponents.URL == nil) {
-            MPLogError(@"invalid notif image URL: %@", imageURLString);
+        NSString *body = object[@"body"];
+        if (![body isKindOfClass:[NSString class]]) {
+            [MPNotification logNotificationError:@"body" withValue:body];
             return nil;
         }
-        imageURL = imageURLComponents.URL;
-    }
-
-    return [[MPNotification alloc] initWithID:ID.unsignedIntegerValue
-                                    messageID:messageID.unsignedIntegerValue
-                                         type:type
-                                        style:style
-                                        title:title
-                                         body:body
-                                 callToAction:callToAction
-                              callToActionURL:callToActionURL
-                                     imageURL:imageURL];
-}
-
-- (instancetype)initWithID:(NSUInteger)ID
-                 messageID:(NSUInteger)messageID
-                      type:(NSString *)type
-                     style:(NSString *)style
-                     title:(NSString *)title
-                      body:(NSString *)body
-              callToAction:(NSString *)callToAction
-           callToActionURL:(NSURL *)callToActionURL
-                  imageURL:(NSURL *)imageURL
-{
-    if (self = [super init]) {
-        if (title.length == 0) {
-            MPLogError(@"Notification title nil or empty: %@", title);
+        
+        NSNumber *bodyColor = object[@"body_color"];
+        if (!([bodyColor isKindOfClass:[NSNumber class]])) {
+            [MPNotification logNotificationError:@"body color" withValue:bodyColor];
             return nil;
         }
-
-        if (body.length == 0) {
-            MPLogError(@"Notification body nil or empty: %@", body);
+        
+        NSNumber *backgroundColor = object[@"bg_color"];
+        if (!([bodyColor isKindOfClass:[NSNumber class]])) {
+            [MPNotification logNotificationError:@"background color" withValue:bodyColor];
             return nil;
         }
-
-        if (!([type isEqualToString:MPNotificationTypeTakeover] || [type isEqualToString:MPNotificationTypeMini])) {
-            MPLogError(@"Invalid notification type: %@, must be %@ or %@", type, MPNotificationTypeMini, MPNotificationTypeTakeover);
-            return nil;
+        
+        NSURL *imageURL = nil;
+        NSString *imageURLString = object[@"image_url"];
+        if (imageURLString != nil && ![imageURLString isKindOfClass:[NSNull class]]) {
+            if (![imageURLString isKindOfClass:[NSString class]]) {
+                [MPNotification logNotificationError:@"image url" withValue:imageURLString];
+                return nil;
+            }
+            
+            NSString *escapedURLString = [imageURLString stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]];
```

Comment:
sanity check in case either the url is not encoded on the sql and we don't encode the url on the decide response. if it's encoded this won't do anything :)